go/vt/vtgate/vindexes/vschema.go

```diff
@@ -207,19 +209,22 @@ func buildTables(source *vschemapb.SrvVSchema, vschema *VSchema) error {
 			if table.Pinned != "" {
 				decoded, err := hex.DecodeString(table.Pinned)
 				if err != nil {
-					return fmt.Errorf("could not decode the keyspace id for pin: %v", err)
+					ksvschema.Error = fmt.Errorf("could not decode the keyspace id for pin: %v", err)
+					goto end
```

Comment:
Instead of this, you can tag the outer loop like this:
```
outer:
  for ksname...
```

and then you can `continue outer`.